solana/bridge_token_factory/programs/bridge_token_factory/src/instructions/admin/initialize.rs

```diff
@@ -0,0 +1,56 @@
+use anchor_lang::prelude::*;
+use wormhole_anchor_sdk::wormhole::{BridgeData, FeeCollector, SequenceTracker};
+
+use crate::{
+    constants::{AUTHORITY_SEED, CONFIG_SEED, DEFAULT_ADMIN},
+    state::config::{Config, ConfigBumps, WormholeConfig},
+    ID,
+};
+
+#[derive(Accounts)]
+pub struct Initialize<'info> {
+    #[account(address = DEFAULT_ADMIN)]
+    pub admin: Signer<'info>,
+
+    #[account(
+        init,
+        payer = payer,
+        space = 8 + Config::INIT_SPACE,
+        seeds = [CONFIG_SEED],
+        bump,
+    )]
+    pub config: Account<'info, Config>,
+
+    pub wormhole_bridge: Account<'info, BridgeData>,
+    pub wormhole_fee_collector: Account<'info, FeeCollector>,
+    pub wormhole_sequence: Account<'info, SequenceTracker>,
+
+    #[account(mut)]
+    pub payer: Signer<'info>,
+
+    pub system_program: Program<'info, System>,
+}
+
+impl<'info> Initialize<'info> {
+    pub fn process(
+        &mut self,
+        derived_near_bridge_address: [u8; 64],
+        config_bump: u8,
+    ) -> Result<()> {
+        let (_, authority_bump) = Pubkey::find_program_address(&[AUTHORITY_SEED], &ID);
+        self.config.set_inner(Config {
+            admin: DEFAULT_ADMIN,
+            derived_near_bridge_address,
+            wormhole: WormholeConfig {
+                bridge: self.wormhole_bridge.key(),
+                fee_collector: self.wormhole_fee_collector.key(),
+                sequence: self.wormhole_sequence.key(),
+            },
+            bumps: ConfigBumps {
+                config: config_bump,
```

Comment:
Saving some CPU by skipping iterating from 255 down to 0 each call (it is required to check the bump is the first one making pubkey out of curve). Not important because we have some CPU per tx + can pay to get more but the check eats unpredicted amount of CPU that's why not perfect

Reply:
Why is authority bump calculated but config bump is passed as an argument? What should we pass for config bump?

Reply:
There are no authority account passed so anchor doesn't know how to calculate. We can pass it, but less accounts in transaction are always better because the max account count is 32 (there are ways to send more but it requires more work)